docs/basic/proxy.mdx

```diff
@@ -21,3 +23,82 @@ setInterval(() => {
   ++state.count
 }, 1000)
 ```
+
+#### Optimizations
+
+Noop
+
+Updates that set the value of a property to the same value are ignored. Subscribers will not be informed of a version change.
+
+```jsx
+const state = proxy({ count: 0 })
+
+state.count = 0 // has no effect
+```
+
+Batching
+
+Multiple changes in the same event loop tick to will be batched together. Subscribers will be notified of a single version change.
+
+```jsx
+const state = proxy({ count: 0, text: 'hello' })
+// subscribers will be notified once after both mutations
+state.count = 1
+state.text = 'world'
+```
+
+#### Nested proxies
+
+Proxies and be nested in other proxy objects and updated as a whole.
+
+```jsx
+import { proxy, useSnapshot } from 'valtio'
+
+const personState = proxy({ name: 'Timo', role: "admin" })
+const authState = proxy({ status: "loggedIn", user: personState })
+
+authState.user.name = 'Nina'
+```
+
+#### Gotchas
+
+You cannot descructure the object from proxy and subscribe to a property.
+
+```jsx
+// don't
+const { user } = proxy({ user: { name: 'Timo' } })
+subscibe(user, () => {...})
+
+// do
+const state = proxy({ user: { name: 'Nina' } })
+subscibe(state.user, () => {...})
```

Comment:
Do you mean `subscribe(state, () => {...})` so that we can detect `state.user = null`-kind-of mutations?